besu/src/main/java/org/hyperledger/besu/cli/util/BesuCommandCustomFactory.java

```diff
@@ -46,4 +45,13 @@ public BesuCommandCustomFactory(final PluginVersionsProvider pluginVersionsProvi
       return constructor.newInstance();
     }
```

Comment:
~~~
final Constructor<T> constructor = cls.getDeclaredConstructor();
    try {
      return constructor.newInstance();
    } catch (Exception e) {
      constructor.setAccessible(true);
      return constructor.newInstance();
    }
~~~
is actually adapted from older (3.1.9) version of defaultFactory's create method. In other words, you don't need to create two method, following would do the trick:

~~~
@Override
  public <T> T create(final Class<T> cls) throws Exception {
    if (CommandLine.IVersionProvider.class.isAssignableFrom(cls)) {
      return (T) new VersionProvider(pluginVersionsProvider);
    }

    return CommandLine.defaultFactory().create(cls);
  }
~~~